metadata_layer.py

```diff
@@ -502,6 +504,128 @@ def __init__(self, content_type, drm_scheme, link=None, rights_uri=None):
             self.rights_uri = self.link.rights_uri
 
 
+class TimestampData(object):
```

Comment:
This is the object itself. Some pieces of information, like `service_type` and `collection_id`, are provided by the superclass service code -- some CoverageProvider subclass can't decide that it's actually a script. 

Other pieces of information, like `start` and `finish` and `counter`, can be provided by the subclassed "do something" method, but the superclass method will provide default values if they're not provided.

Other pieces of information, like `achievements`, are ignored unless the "do something" method sets them.

`exception` is set by the superclass if the "do something" method raises an exception, but theoretically the "do something method" can set it directly and everything should work.